### PR TITLE
Admin advisory management REST API

### PR DESCRIPTION
**Pull request by Yoshani**
## Purpose
This PR introduces a REST API for retrieving the admin advisory banner related configurations.

### Related Issues
- Issue https://github.com/wso2/product-is/issues/15521

### Related PRs
- PR https://github.com/wso2/carbon-identity-framework/pull/4536
- PR https://github.com/wso2/carbon-kernel/pull/3560
- PR https://github.com/wso2/identity-apps/pull/3872
- PR https://github.com/wso2/product-is/pull/15700

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.identity.framework</groupId>
+            <artifactId>org.wso2.carbon.identity.mgt</artifactId>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.common/pom.xml:49]: add a line at the end of file

File: components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.common/src/main/java/org/wso2/carbon/identity/api/server/admin/advisory/management/common/AdminAdvisoryManagementServiceHolder.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+public class AdminAdvisoryManagementServiceHolder {
+
+    private static AdminAdvisoryManagementService adminAdvisoryManagementService;
+
+    public static AdminAdvisoryManagementService getAdminAdvisoryManagementService() {
+
+        return adminAdvisoryManagementService;
+    }
+
+    public static void setAdminAdvisoryManagementService
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.common/src/main/java/org/wso2/carbon/identity/api/server/admin/advisory/management/common/AdminAdvisoryManagementServiceHolder.java:35]: let's add javadoc comments to public methods. fix other places if applicable

File: components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+<!--                    </execution>-->
+<!--                </executions>-->
+<!--                <dependencies>-->
+<!--                    <dependency>-->
+<!--                        <groupId>org.openapitools</groupId>-->
+<!--                        <artifactId>cxf-wso2-openapi-generator</artifactId>-->
+<!--                        <version>1.0.0</version>-->
+<!--                    </dependency>-->
+<!--                </dependencies>-->
+<!--            </plugin>-->
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.v1/pom.xml:58]: let's add a comment describing the purpose of this. Let's ask to run this after a change is done to the swagger contract

File: components/org.wso2.carbon.identity.api.server.admin.advisory.management/pom.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>org.wso2.carbon.identity.api.server.admin.advisory.management</artifactId>
+    <packaging>pom</packaging>
+    <modules>
+        <module>org.wso2.carbon.identity.api.server.admin.advisory.management.common</module>
+        <module>org.wso2.carbon.identity.api.server.admin.advisory.management.v1</module>
+    </modules>
+
+</project>
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.api.server.admin.advisory.management/pom.xml:20]: Add blank line at the end

File: components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.v1/src/main/java/org/wso2/carbon/identity/api/server/admin/advisory/management/v1/core/ServerAdminAdvisoryManagementService.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+        return new ErrorResponse.Builder().withCode(errorMsg.getCode()).withMessage(errorMsg.getMessage())
+                .withDescription(buildErrorDescription(errorMsg, data));
+    }
+
+    private String buildErrorDescription(AdminAdvisoryConstants.ErrorMessage errorEnum, String... data) {
+
+        String errorDescription;
+
+        if (ArrayUtils.isNotEmpty(data)) {
```
Comment (Yoshani) [inline components/org.wso2.carbon.identity.api.server.admin.advisory.management/org.wso2.carbon.identity.api.server.admin.advisory.management.v1/src/main/java/org/wso2/carbon/identity/api/server/admin/advisory/management/v1/core/ServerAdminAdvisoryManagementService.java:90]: simplify if else block

File: components/org.wso2.carbon.identity.api.server.admin.advisory.management/pom.xml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Yoshani) [inline components/org.wso2.carbon.identity.api.server.admin.advisory.management/pom.xml:1]: add license header

**Review by ashensw — APPROVED** [review]

**Pull request merged**